app/models/concerns/firestore_documents.rb

```diff
@@ -72,13 +79,30 @@ def self.by_study_and_name(accession, name)
       doc_ref.any? ? self.new(doc_ref.first) : nil
     end
 
+    def self.by_study_and_file_id(accession, file_id)
+      documents = self.query_by(study_accession: accession, file_id: file_id)
+      documents.map {|doc| self.new(doc)}
+    end
+
     # shortcut method to determine if there are any documents of this type for a study
     # useful for methods like @study.has_expression_data?
     # will limit to only a single document to cut down on query costs
     def self.study_has_any?(accession)
       self.query_by(study_accession: accession).any?
     end
 
+    ##
+    # Delete methods
+    ##
+
+    def self.delete_by_study(accession, threads=10)
+      delete_documents(self.query_by(study_accession: accession), threads)
+    end
+
+    def self.delete_by_study_and_file(accession, file_id, threads=10)
```

Comment:
Whats the reason for self.delete_by_study and this one?

Reply:
Mainly, we would use `delete_by_study_and_file` when a user deletes a single file from their study.  However, in the case of a user deleting their entire study without first clearing out files, we could use `delete_by_study` instead of having to loop through each file first.